### ISSUE TITLE
Untrusted pointer dereference in 16 "TEE_xx" functions.

### ISSUE THREAD

**Issue by SyzTrust** [Samsung/mTower#93]
Hi there,
Recently, we found several untrusted pointer dereference bugs in the following functions:
1. TEE_MACComputeFinal
2. TEE_SetOperationKey2
3. TEE_MACUpdate
4. TEE_GetOperationInfoMultiple
5. TEE_AEEncryptFinal, TEE_MACInit
6. TEE_SetOperationKey
7. TEE_ResetOperation
8. TEE_DigestUpdate
9. TEE_AEDecryptFinal
10. TEE_CipherInit
11. TEE_FreeOperation
12. TEE_DigestDoFinal
13. TEE_AllocateOperation
14. TEE_FreeTransientObject
15. TEE_AEUpdate
16. TEE_ResetOperation

Here we use " `function name` " to present any of the above functions.

**Affected product(s)/code base**

mTower m2351 v0.4.0

**Attack vector(s)**

To exploit the vulnerability, invoke the function " `function name` " and pass an invalid pointer to the first parameter (TEE_OperationHandler).

**Suggested description of the vulnerability for use in the CVE**

Untrusted pointer dereference vulnerability in the" `function name` " function in Tsinglink Cloud TinyTEE M2351 v3.0.1 allows a trusted application to trigger a Denial of Service (DoS) via invoking the function " `function name` " with an invalid pointer of the first parameter, that is, TEE_OperationHandle.

**Discoverer(s)/Credits** 

SyzTrust

**Additional context**
The " function name " function does not check whether the first parameter is in trusted space or not. When invoking this function and passing an invalid pointer, it will dereference the address and crash the trusted execution environment kernel, resulting in a Denial of Service (DoS).

Our suggested solution is to: add a  handle list to record the handles that are allocated and returned by TEE_AlloacateOperation() and add the check on the TEE_OperationHandle to see if they are on the handle list. Once the TEE_OperationHandle is released, then delete the handle in the handle list.
Or, when trying to dereference any pointer in mTower, add a check to see if its address is in the legal memory region. For instance, the Linux kernel always checks if the userspace pointer is valid.
```C
static inline void __chk_user_ptr(const volatile void *p, size_t siez){
    assert(p >= __user_addr_min && p + size <= __user_addr_max);
}
```

Looking forward to your reply!

**Comment by tdrozdovsky** [comment]
Will be reviewed soon

**Comment by tdrozdovsky** [comment]
Fixed commit 3002548120161953559be92baa3da121473cc6b5
